src/main/java/it/okkam/validation/FiscalCodeValidator.java

```diff
@@ -155,22 +161,16 @@
   /**
    * Calculate valid italian fiscal codes for given data.
    * 
-   * @param conf
-   *          the FiscalCodeConf
-   * @param surname
-   *          person surname
-   * @param name
-   *          person name
-   * @param birthDate
-   *          person birth date (as dd/MM/yyyy)
-   * @param townOfBirth
-   *          person town of birth
-   * @param gender
-   *          person gender
+   * @param conf the FiscalCodeConf
+   * @param surname person surname
+   * @param name person name
+   * @param birthDate person birth date (as dd/MM/yyyy)
+   * @param townOfBirth person town of birth
+   * @param gender person gender
    * @return the list of valid fiscal codes, null if the fiscal code cannot be computed
    */
   public static String[] calcoloCodiceFiscale(FiscalCodeConf conf, String surname, String name,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 51 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://dev.okkam.it/sonar/coding_rules#rule_key=squid%3AS3776)